flow360/component/volume_mesh.py

```diff
@@ -452,8 +516,29 @@ def _submit_upload_mesh(self, progress_callback=None):
         info = VolumeMeshMeta(**resp)
         self._id = info.id
         mesh = VolumeMesh(self.id)
-        mesh.upload_file(remote_file_name, self.file_name, progress_callback=progress_callback)
-        mesh._complete_upload(remote_file_name)
+
+        # parallel compress and upload
+        if (
+            original_compression == CompressionFormat.NONE
+            and self.compress_method == CompressionFormat.BZ2
+        ):
+            upload_id = mesh.create_multipart_upload(remote_file_name)
+            self._compress_and_upload_chunks(upload_id, mesh, remote_file_name)
+            mesh._complete_upload(remote_file_name)
+        else:
+            if original_compression == CompressionFormat.NONE:
+                compressed_file_name = self._zstd_compress(self.file_name)
+                compressed_file_name = (
+                    compressed_file_name if compressed_file_name is not None else self.file_name
+                )
```

Comment:
If we upload uncompressed file here, but in `NewVolumeMeshRequest()` we already provided zst, it will error out. So we need to assume `compressed_file_name` exists.